.github/workflows/security-code-scanner.yml

```diff
@@ -0,0 +1,29 @@
+name: 'MetaMask Security Code Scanner'
+
+on:
+  push:
+    branches: ['develop']
+  pull_request:
+    branches: ['develop']
+
+jobs:
+  run-security-scan:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    steps:
+      - name: MetaMask Security Code Scanner
+        uses: MetaMask/Security-Code-Scanner@main
+        with:
+          repo: ${{ github.repository }}
+          paths_ignored: |
+            tests/
+            docs/
+            '**/*.test.js'
+            '**/*.test.ts'
+            node_modules
+          rules_excluded: example
```

Comment:
Is there a way to set up a config/ignore file for this?

Reply:
- `merged-packages/` is used as a temporary directory for package migrations and should be ignored. Incoming packages can be scanned once the migration process is completed.
- `**/test/`: `packages/transactions-controller/test`
- The only `js` files in the repo are config files. At the very least, `**/jest.config.js`, `**/jest.environment.js` should be ignored.

Reply:
Looks like keeping `scripts/` shouldn't be disruptive based on the scan results. `package-template/` has been moved under `scripts/` so no problem leaving that out as well.

Reply:
Added a revised ignore list above: https://github.com/MetaMask/core/pull/3929#discussion_r1495962550

Reply:
Could we also replace `tests/` with `**/tests/`? We want both `**/tests/` and `**/test/`, since they're both used depending on the package. Half of the alerts we're seeing right now seem to be from mock urls found under `**/tests/`.